README.md

```diff
@@ -13,20 +13,37 @@ for saving storage space and keeping repositories up-to-date.
 
 ## Key concepts
 
- - **Borges producer**: a standalone process that reads repository URLs (from RabbitMQ or file) and schedules fetching this repository.
+### Borges producer
 
- - **Borges consumer**: a standalone process that takes URLs from RabbitMQ, clones remote repository and pushes it to the appropriate *Rooted Repository* in the storage (local filesystem or HDFS).
+A standalone process that reads repository URLs (from RabbitMQ or file) and schedules fetching this repository.
 
- - **Borges packer**: a standalone process that takes repository paths (or URLs) from a file and packs them into siva files (as a *Rooted Repository*) in the given output directory.
+### Borges consumer
 
- - **Rooted Repository**: a standard Git repository that stores all objects from all repositories that share common history, identified by same initial commit. It is stored in [Siva](https://github.com/src-d/go-siva) file format.
+A standalone process that takes URLs from RabbitMQ, clones remote repository and pushes it to the appropriate *Rooted Repository* in the storage (local filesystem or HDFS).
 
-   ![Root Repository explanatory diagram](https://user-images.githubusercontent.com/5582506/30617179-2aba194a-9d95-11e7-8fd5-0a87c2a595f9.png)
+### Borges packer
 
-Consumer and Producer run independently, communicating though a RabbitMQ instance
-and storing repository meta-data in PostgreSQL.
+A standalone process that takes repository paths (or URLs) from a file and packs them into siva files (as a *Rooted Repository*) in the given output directory.
 
-Packer does not need a RabbitMQ or a PostgreSQL instance and is not meant to be used as a pipeline, that's what consumer and producer are meant for.
+### Rooted Repository
+
+A rooted repository is a [bare Git repository](http://www.saintsjd.com/2011/01/what-is-a-bare-git-repository/) that stores all objects from all repositories that share a common history, that is, they have the same initial commit. It is stored using the [Siva](https://github.com/src-d/go-siva) file format.
+
+![Root Repository explanatory diagram](https://user-images.githubusercontent.com/5582506/30617179-2aba194a-9d95-11e7-8fd5-0a87c2a595f9.png)
+
+Rooted repositories have a few particularities that you should know to work with them effectively:
+
+- They have no `HEAD` reference.
+- All references are of the following form: `{REFERENCE_NAME}/{REMOTE_NAME}`. For example, the reference `refs/heads/master` of the remote `foo` would be `/refs/heads/master/foo`. The remote name in rooted repositories generated by borges is always the `id` (in `UUID` form) of the repository in the PostgreSQL database.
```

Comment:
Is the remote URL in config of the repo itself?

Reply:
yup

Reply:
That would probably deserve a note.

Reply:
If I understood what you meant correctly, that's in the next line